lib/runtime/invalid_transaction.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+import (
+	"errors"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// InvalidTransaction is child VDT of TransactionValidityError
+type InvalidTransaction scale.VaryingDataType
+
+// Index fulfils the VaryingDataTypeValue interface.  T
+func (InvalidTransaction) Index() uint {
+	return 0
+}
+
+var (
+	errUnexpectedTxCall         = errors.New("call of the transaction is not expected")
+	errInvalidPayment           = errors.New("invalid payment")
+	errInvalidTransaction       = errors.New("invalid transaction")
+	errOutdatedTransaction      = errors.New("outdated transaction")
+	errBadProof                 = errors.New("bad proof")
+	errAncientBirthBlock        = errors.New("ancient birth block")
```

Comment:
If the only place these errors are used is to call `.Error()` on them below, just inline the strings directly in `func (i *InvalidTransaction) Error() string {` to reduce the Go API.

Reply:
I did this to align our naming (and have representations of) substrate errors. I can do this, but I think it would lose a bit of clarity. Also, some of these are used in the codebase in other places, so I think a place of further refactor could be to define these exported errors elsewhere.

That being said, I can still make this change if you think its best I just liked the naming clarity

Reply:
Yeah for unexported errors only used as strings, definitely remove them and inline the strings directly where needed.